### PR TITLE
:art: add topics button type and change color of text in primary ligh…

### LINKED ISSUE #324: text color in buttons
Capture d’écran de 2022-11-09 12-07-59

I think the buttons will have a better render if the font was white on the red button. What do you think about that ?

### PR DESCRIPTION
**Pull request by MailyLehoux**
resolve #324

### CODE REVIEW

File: src/components/Button.tsx [hunk 1]
```diff
@@ -36,9 +40,19 @@ const Button = ({
...
         style === "outlined" &&
         `text-${type === "primary" ? "dark-red" : "white"}`
-      } text-base font-bold py-2 px-5 rounded-full disabled:opacity-25`}
+      } text-base ${style != "faded" && "font-bold"} py-${
+        type === "tertiary" ? "1" : "2"
+      } px-5 rounded-full disabled:opacity-25 ${
+        style === "contained" &&
+        type === "tertiary" &&
+        `gradient-${color && color.length > 0 ? color : "red"}`
+      } ${style === "faded" && type === "tertiary" && "gradient-red-faded"}`}
```
Comment (benjaminbrgs) [inline src/components/Button.tsx:55]:
```suggestion
      } ${style === "faded" && type === "tertiary" && `gradient-${color && color.length > 0 ? color : "red"}-faded`}`}
```

Comment (benjaminbrgs) [reply src/components/Button.tsx:55]: Color props is missing on this line

**Review by benjaminbrgs — APPROVED** [review]
Nice work ! Buttons are much clearer now 💯

File: src/components/Button.tsx [hunk 2]
```diff
@@ -18,11 +19,14 @@ const Button = ({
...
+  color,
 }: ButtonProps) => {
   return (
     <button
       className={`${
-        type === "primary" && style === "contained" && "gradient-red"
+        type === "primary" &&
+        style === "contained" &&
+        `gradient-${color && color.length > 0 ? color : "red"}`
       } ${
```
Comment (PierreBrethes) [inline src/components/Button.tsx:30]: I think you should remove all this css and put it just above in a function that returns css. Using switches, cause right now it's not really readable and scalable (even if it works)

File: src/components/Topics.tsx [hunk 3]
```diff
@@ -33,24 +34,24 @@ const Topics = ({
...
-              } ${readOnly ? "cursor-default" : ""}`}
-              onClick={
-                selected
-                  ? () => removeCallback(topic)
-                  : () => addCallback(topic)
-              }
-            >
-              <span className="px-2 py-1 text-white">{topic.name}</span>
-            </button>
+            <div className="flex-inital py-2" key={key}>
```
Comment (PierreBrethes) [inline src/components/Topics.tsx:41]: Just a tip, when you put key={key}, it's not really good. As in your html you will have 0,1,2,3,etc... But elsewhere in the app you can have the same. Imagine you put 2 components with the same key={key} then you'll have two times html elements with same key. 
The trick is just to put for an exemple key={`topic-${key}`} and then it's more readable, no duplicates, and happy html scroller :D

**Review by PierreBrethes — APPROVED** [review]
Great job overwall tho 🚀

File: src/components/Button.tsx (hunk 2 above)
Comment (MailyLehoux) [reply src/components/Button.tsx:30]: Good idea. can be good to fix it in a next issue

**Pull request merged**